File: src/main/java/net/szymonsawicki/reactivetimesheetapp/domain/user/dto/GetUserDto.java
package net.szymonsawicki.reactivetimesheetapp.domain.user.dto;

import net.szymonsawicki.reactivetimesheetapp.domain.user.User;
import net.szymonsawicki.reactivetimesheetapp.domain.user.type.Role;

public record GetUserDto(String id, String username, String password, Role role, String teamId) {
    public User toUser() {
        return User.builder()
                .id(id)
                .username(username)
                .password(password)
                .role(role)
                .teamId(teamId)
                .build();
    }
}

File: frontend/src/model/GetUserDto.js
export default class GetUserDto {
    constructor(id, username, role, teamName) {
        this.id = id;
        this.username = username;
        this.role = role;
        this.teamName = teamName;
    }

    toString() {
        return `${this.id} ${this.username} ${this.role} ${this.teamName}`;
    }
}